### DISCUSSION TITLE
fail to circularize the cp genome assembly

### DISCUSSION THREAD

**Discussion by IntronQ** [Kinggerm/GetOrganelle#224] [Q&A]
I have tried the different augments to assemble the cp genome, but all the results are broken assembly graph .
I'm trying to assemble the raw reads directly using SPAdes, but I have no idea which SPAdes outputs are chosen for the next step and how to use getOrganelle (command line):  use get_organelle_from_assembly.py to extract the target organelle genome sequence. 
[get_org.log.txt](https://github.com/Kinggerm/GetOrganelle/files/10314785/get_org.log.txt)
![1672238962955](https://user-images.githubusercontent.com/121562754/209829881-396c5bef-a471-4b8d-b1dd-cd2cde1b40d0.jpg)

**Comment by Kinggerm** [comment]
For the current GetOrganelle versions (<=1.7.7), you should use `<spades_output_dir>/assembly_graph.fastg` as the input of `get_organelle_from_assembly.py`. 

Meanwhile, `get_organelle_from_assembly.py` can be able to utilize the scaffolding information only when the result is incomplete. This is disabled by default (see the option `--spades-out-dir`). To enable this, you have to input the spades output directory containing two files: `scaffolds.fasta` and `scaffolds.paths` using the flag `--spades-out-dir`, e.g. `--spades-out-dir my_spades_dir`. Please note that `<spades_output_dir>/assembly_graph_with_scaffolds.gfa` is currently not exploited by GetOrganelle.

**Reply by IntronQ** [reply]
Thanks for you replying!  Here comes another problem.   The command I used : <get_organelle_from_assembly.py -g dp_output/assembly_graph.fastg -F embplant_pt -o output --min-depth 10>,  and the output information shows:

GetOrganelle v1.7.6.1

get_organelle_from_assembly.py isolates organelle genomes from assembly graph.
Find updates in https://github.com/Kinggerm/GetOrganelle and see README.md for more information.

Python 3.8.13 (default, Mar 28 2022, 11:38:47)  [GCC 7.5.0]
PLATFORM: Linux pbi 4.18.0-373.el8.x86_64 #1 SMP Tue Mar 22 15:11:47 UTC 2022 x86_64 x86_64
PYTHON LIBS: GetOrganelleLib 1.7.6.1; numpy 1.23.1; sympy 1.10.1; scipy 1.9.1
blastn not accessible!

I use conda to install the GetOrganelle. Can you help to fix this problem?

**Reply by Kinggerm** [reply]
Did you use a different GetOrganelle environment? According to the previous log, you should have blastn installed.

**Reply by IntronQ** [reply]
I use "conda activate “ to implement the script, and it works! But, there still comes the incomplete result.  Do you have any suggestions for this situation？Many thanks!

Here is the command line I used:
<get_organelle_from_assembly.py -g dapiao_6_spades/dapiao_output/assembly_graph.fastg -F embplant_pt -o dapiao_9 --min-depth 10>
< join_spades_fastg_by_blast.py -g slimmed_assembly_graph.fastg -f ref.fasta >

The final output graph shows below.

![1672363216712](https://user-images.githubusercontent.com/121562754/210025437-04b90f31-fc41-486d-85eb-8bee9bf626e9.jpg)
[slimmed_assembly_graph.slim.log.txt](https://github.com/Kinggerm/GetOrganelle/files/10322726/slimmed_assembly_graph.slim.log.txt)
[get_org.log.txt](https://github.com/Kinggerm/GetOrganelle/files/10322727/get_org.log.txt)

**Reply by Kinggerm** [reply]
The coverage seems high enough. I'm not sure of the reasons for the incomplete. Maybe you can try Velvet instead of SPAdes to get the assembly graph then use get_organelle_from_assembly.py to extract.

If it still fails, 
You may follow the information in the log file
>2022-12-30 09:04:57,920 - INFO: If the result is nearly complete, you can try join_spades_fastg_by_blast.py to fill N-gaps in-between contigs with a closely-related reference.
2022-12-30 09:04:57,920 - INFO: If you have questions for us, please provide us with the get_org.log.txt file and the post-slimming graph in the format you like!

It is also stated in the FAQ. You may check it out.
